src/libraries/Common/src/System/Diagnostics/NetFrameworkUtils.cs

```diff
@@ -22,14 +22,17 @@ internal static void EnterMutex(string name, ref Mutex mutex)
 
         internal static void EnterMutexWithoutGlobal(string mutexName, ref Mutex mutex)
         {
-            Mutex tmpMutex = new Mutex(false, mutexName, out _);
-
-            // Specify a SID in case the mutex has not yet been created; this prevents it from using the SID from the current thread.
-            // This SID (AuthenticatedUserSid) is the same one used by .NET Framework.
-            MutexSecurity sec = new MutexSecurity();
-            SecurityIdentifier authenticatedUserSid = new SecurityIdentifier(WellKnownSidType.AuthenticatedUserSid, null);
-            sec.AddAccessRule(new MutexAccessRule(authenticatedUserSid, MutexRights.Synchronize | MutexRights.Modify, AccessControlType.Allow));
-            tmpMutex.SetAccessControl(sec);
+            Mutex tmpMutex = new Mutex(false, mutexName, out bool createdNew);
+
+            if (createdNew)
+            {
+                // Specify a SID in case the mutex has not yet been created; this prevents it from using the SID from the current thread.
+                // This SID (AuthenticatedUserSid) is the same one used by .NET Framework.
+                MutexSecurity sec = new MutexSecurity();
+                SecurityIdentifier authenticatedUserSid = new SecurityIdentifier(WellKnownSidType.AuthenticatedUserSid, null);
+                sec.AddAccessRule(new MutexAccessRule(authenticatedUserSid, MutexRights.Synchronize | MutexRights.Modify, AccessControlType.Allow));
+                tmpMutex.SetAccessControl(sec);
```

Comment:
See https://github.com/dotnet/runtime/issues/120085 for context on why MutexAcl.Create wouldn't work.

Reply:
I see, do we introduce any sort of race condition here?  Can some other thread/process do something to interrupt between the creation of the mutex and our changing of the ACL such that it prevents our changing the ACL?

IIRC we added all the creation static methods to address race conditions like that.

Reply:
Per offline discussion - the race condition was here already and solving #120085 is too risky for .NET 10.0.  This change is minimal to resolve the regression in .NET 10.0.